File: DP/1912 - 연속합.py
import sys
read = sys.stdin.readline

n = int(read())
a = list(map(int, read().split()))

dp = [a[0]]

for i in range(n-1) :
    dp.append(max(dp[i]+a[i+1], a[i+1]))

'''
for i in range(n) :
    dp.append(a[i])
    for j in range(i+1, n) :
        a[i] = a[i] + a[j]
        dp[i] = max(dp[i], a[i])
'''

print(max(dp))

File: DP/1912 - 연속합.js
const fs = require('fs');
const input = fs
  .readFileSync('./DP/1912input.txt')
  .toString()
  .trim()
  .split('\r\n');
// const input = fs.readFileSync('dev/stdin').toString().trim().split('\n');

const n = +input[0];
const arr = input[1].split(' ').map(Number);

const dp = [arr[0]];

for (let i = 1; i < n; i++) {
  dp.push(Math.max(dp[i - 1] + arr[i], arr[i]));
}

console.log(Math.max(...dp));
